invalid memory address or nil pointer dereference on `rad credential show azure`

### Steps to reproduce

```
$ rad credential register azure sp --client-id $AZURE_CLIENT_ID  --client-secret $AZURE_CLIENT_SECRET  --tenant-id $AZURE_TENANT_ID
Registering credential for "azure" cloud provider in Radius installation "Kubernetes (context=nimble)"...
Successfully registered credential for "azure" cloud provider. Tokens may take up to 30 seconds to refresh.
                                                                                                                                
$ rad credential list
Listing credentials for all cloud providers for Radius installation "Kubernetes (context=nimble)"...
PROVIDER  REGISTERED
azure     true

$ rad credential show azure
Showing credential for cloud provider "azure" for Radius installation "Kubernetes (context=nimble)"...
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x1062544a4]

goroutine 1 [running]:
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End.deferwrap1()
        /home/runner/go/pkg/mod/go.opentelemetry.io/otel/sdk@v1.35.0/trace/span.go:467 +0x2c
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End(0x14000333680, {0x0, 0x0, 0x0?})
        /home/runner/go/pkg/mod/go.opentelemetry.io/otel/sdk@v1.35.0/trace/span.go:500 +0x99c
panic({0x106a4c880?, 0x1086953e0?})
        /opt/hostedtoolcache/go/1.24.1/x64/src/runtime/panic.go:792 +0x124
github.com/radius-project/radius/pkg/cli/cmd/credential/show.(*Runner).Run(0x14000836d20, {0x106f215d0, 0x140008478f0})
        /home/runner/work/radius/radius/pkg/cli/cmd/credential/show/show.go:134 +0x4c4
github.com/radius-project/radius/pkg/cli/cmd/credential/show.NewCommand.RunCommand.func2(0x1400084cf08, {0x14000467560?, 0x4?, 0x1062ef687?})
        /home/runner/work/radius/radius/pkg/cli/framework/framework.go:192 +0x7c
github.com/spf13/cobra.(*Command).execute(0x1400084cf08, {0x14000467530, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0x828
github.com/spf13/cobra.(*Command).ExecuteC(0x1086ca4a0)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1064
github.com/radius-project/radius/cmd/rad/cmd.Execute()
        /home/runner/work/radius/radius/cmd/rad/cmd/root.go:169 +0x1d0
main.main()
        /home/runner/work/radius/radius/cmd/rad/main.go:26 +0x1c
```

### Observed behavior

_No response_

### Desired behavior

Show my service principal ID

### Workaround

_No response_

### rad Version

$ rad version              
CLI Version Information:
RELEASE   VERSION   BICEP     COMMIT
edge      75372b5   0.35.1    75372b5dc51b4b396eeda4b9d95af25d210054fa

Control Plane Information:
STATUS     VERSION
Installed  0.42.42-dev

### Operating system

_No response_

### Additional context

_No response_

### Would you like to support us?

- [ ] Yes, I would like to support you

[AB#15779](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/15779)

--- a/pkg/cli/cmd/credential/show/show_test.go
+++ b/pkg/cli/cmd/credential/show/show_test.go
@@ -151,6 +151,7 @@ func Test_Run(t *testing.T) {
 			}
 			require.Equal(t, expected, outputSink.Writes)
 		})
+        
 		t.Run("Azure WorkloadIdentity - Exists", func(t *testing.T) {
 			ctrl := gomock.NewController(t)
 

--- a/pkg/cli/cmd/credential/show/show_test.go
+++ b/pkg/cli/cmd/credential/show/show_test.go
@@ -199,6 +199,7 @@ func Test_Run(t *testing.T) {
 			}
 			require.Equal(t, expected, outputSink.Writes)
 		})
+		
 		t.Run("Not Found", func(t *testing.T) {
 			ctrl := gomock.NewController(t)
 

--- a/pkg/cli/cmd/credential/show/show_test.go
+++ b/pkg/cli/cmd/credential/show/show_test.go
@@ -151,6 +151,53 @@ func Test_Run(t *testing.T) {
 			}
 			require.Equal(t, expected, outputSink.Writes)
 		})
+		t.Run("Azure WorkloadIdentity - Exists", func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+
+			provider := cli_credential.ProviderCredentialConfiguration{
+				CloudProviderStatus: cli_credential.CloudProviderStatus{
+					Name:    "azure",
+					Enabled: true,
+				},
+				AzureCredentials: &cli_credential.AzureCredentialProperties{
+					Kind: to.Ptr("WorkloadIdentity"),
+				},
+			}
+
+			client := cli_credential.NewMockCredentialManagementClient(ctrl)
+			client.EXPECT().
+				Get(gomock.Any(), "azure").
+				Return(provider, nil).
+				Times(1)
+
+			outputSink := &output.MockOutput{}
+
+			runner := &Runner{
+				ConnectionFactory: &connections.MockFactory{CredentialManagementClient: client},
+				Output:            outputSink,
+				Workspace:         &workspaces.Workspace{Connection: connection},
+				Kind:              "azure",
+				Format:            "table",
+			}
+
+			err := runner.Run(context.Background())
+			require.NoError(t, err)
+
+			credentialFormatOutput := credentialFormatAzureWorkloadIdentity()
+
+			expected := []any{
+				output.LogOutput{
+					Format: "Showing credential for cloud provider %q for Radius installation %q...",
+					Params: []any{"azure", "Kubernetes (context=my-context)"},
+				},
+				output.FormattedOutput{
+					Format:  "table",
+					Obj:     provider,
+					Options: credentialFormatOutput,
+				},
+			}
+			require.Equal(t, expected, outputSink.Writes)
+		})
 		t.Run("Not Found", func(t *testing.T) {
 			ctrl := gomock.NewController(t)
 

--- a/pkg/cli/credential/azure_credential_management.go
+++ b/pkg/cli/credential/azure_credential_management.go
@@ -138,6 +138,7 @@ func (cpm *AzureCredentialManagementClient) Get(ctx context.Context, credentialN
 				Enabled: true,
 			},
 			AzureCredentials: &AzureCredentialProperties{
+				Kind: (*string)(azureServicePrincipal.Kind),
 				ServicePrincipal: &AzureServicePrincipalCredentialProperties{
 					ClientID: azureServicePrincipal.ClientID,
 					Kind:     (*string)(azureServicePrincipal.Kind),
@@ -159,6 +160,7 @@ func (cpm *AzureCredentialManagementClient) Get(ctx context.Context, credentialN
 				Enabled: true,
 			},
 			AzureCredentials: &AzureCredentialProperties{
+				Kind: (*string)(azureWorkloadIdentity.Kind),
 				WorkloadIdentity: &AzureWorkloadIdentityCredentialProperties{
 					ClientID: azureWorkloadIdentity.ClientID,
 					Kind:     (*string)(azureWorkloadIdentity.Kind),

